src/Observer.ts

```diff
@@ -0,0 +1,190 @@
+        if (this.format === "json") {
+            // 如果websocket中没有senObj就添加这个方法对象
+            if (!("sendObj" in (this.WebSocket as WebSocket))) {
+                // 将发送的消息转为json字符串
+                (this.WebSocket as WebSocket).sendObj = (obj: JSON) => (this.WebSocket as WebSocket).send(JSON.stringify(obj));
+            }
+        }
+        return this.WebSocket;
+    }
+    // 重新连接
+    reconnect(): void {
+        // 已重连次数小于等于设置的连接次数时执行重连
+        if (this.reconnectionCount <= this.reconnectionAttempts) {
+            this.reconnectionCount++;
+            // 清理上一次重连时的定时器
+            clearTimeout(this.reconnectTimeoutId);
+            // 开始重连
+            this.reconnectTimeoutId = setTimeout(() => {
+                // 如果启用vuex就触发vuex中的重连方法
+                if (this.store) {
+                    this.passToStore("SOCKET_RECONNECT", this.reconnectionCount);
+                }
+                // 重新连接
+                this.connect(this.connectionUrl, this.opts);
+                // 触发WebSocket事件
+                this.onEvent();
+            }, this.reconnectionDelay);
+        } else {
+            // 如果启用vuex则触发重连失败方法
+            if (this.store) {
+                this.passToStore("SOCKET_RECONNECT_ERROR", true);
+            }
+        }
+    }
+
+    // 事件分发
+    onEvent(): void {
+        ["onmessage", "onclose", "onerror", "onopen"].forEach((eventType: string) => {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            (this.WebSocket as WebSocket)[eventType] = (event) => {
+                Emitter.emit(eventType, event);
+
+                // 调用vuex中对应的方法
+                if (this.store) {
+                    this.passToStore("SOCKET_" + eventType, event);
+                }
+
+                // 处于重新连接状态切事件为onopen时执行
+                if (this.reconnection && eventType === "onopen") {
+                    // 设置实例
+                    this.opts.$setInstance && this.opts.$setInstance(event.currentTarget);
+                    // 清空重连次数
+                    this.reconnectionCount = 0;
+                }
+
+                // 如果处于重连状态且事件为onclose时调用重连方法
+                if (this.reconnection && eventType === "onclose") {
+                    this.reconnect();
+                }
+            };
+        });
+    }
+
+    /**
+     * 触发vuex中的方法
+     * @param eventName 事件名称
+     * @param event 事件
+     */
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    passToStore(eventName: string, event: any): void {
+        // 如果参数中有传事件处理函数则执行自定义的事件处理函数，否则执行默认的处理函数
+        if (this.passToStoreHandler) {
+            this.passToStoreHandler(eventName, event, this.defaultPassToStore.bind(this));
+        } else {
+            this.defaultPassToStore(eventName, event);
+        }
+    }
+
+    /**
+     * 默认的事件处理函数
+     * @param eventName 事件名称
+     * @param event 事件
+     */
+    defaultPassToStore(
+        eventName: string,
+        event: {
```

Comment:
create new type for this

Reply:
Create a new type? what type?

Reply:
how about `VuexStore` since that's really what this property is defining